docs/react-v9/contributing/rfcs/shared/build-system/stop-styles-transforms.md

```diff
@@ -0,0 +1,208 @@
+
+<details>
+  <summary>Pre-processed code</summary>
+
+```ts
+// @fluentui/react-menu/lib/components/MenuDivider/useMenuDividerStyles.styles.js
+// 📝 output is simpfied
+
+const useStyles = /*#__PURE__*/ __styles(
+  {
+    root: {
+      B6of3ja: 'fvjh0tl',
+      t21cq0: ['f1rnx978', 'f1q7jvqi'],
+    },
+  },
+  {
+    d: ['.fvjh0tl{margin-top:4px;}', '.f1rnx978{margin-right:-5px;}', '.f1q7jvqi{margin-left:-5px;}'],
+  },
+);
+```
+
+</details>
+
+Addressing prefixing for pre-processed styles presents a challenge, as they have already been transformed into CSS rules and classes:
+
+- 👎 **Webpack loader** (i.e. build time) - Will make loaders more complex and less efficient
+- 👎 **Runtime** - Will require additional logic in `__styles()` (_artifact of AOT compilation in Griffel_) to prefix classes that makes the idea of AOT compilation obsolete
+
+## Current workaround
+
+We have proposed a temporary workaround in [microsoft/griffel#453](https://github.com/microsoft/griffel/issues/453#issuecomment-1850115159). The workaround involves increasing the specificity of the CSS rules in the consuming app. This is achieved by adding a unique class to the root element of the app, for example:
+
+```css
+.color-red {
+}
+/* becomes ⬇️ */
+.PREFIX .color-red {
+}
+```
+
+As a result, the CSS rules will have a higher specificity and will override any conflicting rules from the application. This approach is not ideal, as it affects performance (makes CSS rules less efficient).
+
+> Note: We cannot prefix classes as Fluent components have pre-processed styles:
+>
+> ```js
+> const useStyles = /*#__PURE__*/ __styles(
+>   // Part 1: CSS classes mapping
+>   { root: { B6of3ja: 'fvjh0tl' } },
+>   // Part 2: CSS rules
+>   { d: ['.fvjh0tl{margin-top:4px;}'] },
+> );
+> ```
+>
+> - We cannot prefix classes in the mapping with current APIs (part 1) as classes have been already hashed~~~~
+> - We can prefix CSS classes (part 2) properly only but invoking Stylis (CSS preprocessor) on every CSS rule
+
+> Note: `__styles()` is a result of AOT compilation in Griffel, adding additional responsibilities to it makes the idea of AOT compilation obsolete.
+
+A [StackBlitz example](https://stackblitz.com/edit/vitejs-vite-d11ccd) demonstrates this workaround.
+
+## Proposal
+
+### Option B: Ship ESM output with unprocessed styles
+
+This option is more complex to maintain but ensures backward compatibility and avoids breaking changes.
+
+We will offer three outputs for each package under the following `exports` in `package.json`:
+
+- `"import": "./lib/index.js"` (as is) - ESM output with pre-processed styles (default for bundlers)
+- `"require": "./lib-commonjs/index.js"` (as is) - CJS output with pre-processed styles (default for Node.js)
+- `"import-raw": "./lib-raw/index.js"` (🆕)- ESM output with unprocessed styles
+
+Consumers desiring to use the class prefixing feature will need to configure their bundler to use the `"import-raw"` output. For example, in Webpack:
```

Comment:
With this approach can we, if we major rev would it be reasonable to switch import-raw to the be the default?

Reply:
I dont think this is feasible in order to follow with our semver and support commitment following our LTS release cycle. probably after 2 major bumps we could remove this completely and switch to only follow standard APIs, but as I mentioned in my summary we should avoid introducing this artificial API.